index.js

```diff
@@ -1,15 +1,65 @@
-const port = parseInt(process.env.PORT, 10) || 3000;
 const app = require("express")();
 const bodyParser = require("body-parser");
 const xrefResponseBody = require("respec-xref-route");
-const xrefData = require("./xref-data.json");
+const crypto = require("crypto");
+const { exec } = require("child_process");
 
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(bodyParser.json());
+let xrefData = require("./xref-data.json");
+
+const conf = {
+  port: parseInt(process.env.PORT, 10) || 3000,
+  bikeshedSecret: process.env.BIKESHED_SECRET || "",
+};
+
+app.use(bodyParser.urlencoded({ verify: rawBodyParser, extended: false }));
+app.use(bodyParser.json({ verify: rawBodyParser }));
 
 app.post("/xref", (req, res) => {
   const body = xrefResponseBody(req.body, xrefData);
   res.json(body);
 });
 
-app.listen(port, () => console.log(`Listening on port ${port}!`));
+app.post("/xref/update", handleBikeshedUpdate);
+
+app.listen(conf.port, () => console.log(`Listening on port ${conf.port}!`));
+
+function rawBodyParser(req, res, buf, encoding) {
+  if (buf && buf.length) {
+    req.rawBody = buf.toString(encoding || "utf8");
+  }
+}
+
+function handleBikeshedUpdate(req, res) {
```

Comment:
Let's make a new folder called "routes" and move all this stuff into "xref-update-route.js".

Reply:
Yeah, that might be nice.